spring-test/src/main/java/org/springframework/test/web/client/match/FormDataRequestMatchers.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2002-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.test.web.client.match;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.hamcrest.Matcher;
+import org.hamcrest.Matchers;
+
+import org.springframework.http.client.ClientHttpRequest;
+import org.springframework.http.converter.FormHttpMessageConverter;
+import org.springframework.mock.http.MockHttpInputMessage;
+import org.springframework.mock.http.client.MockClientHttpRequest;
+import org.springframework.test.web.client.RequestMatcher;
+import org.springframework.util.Assert;
+import org.springframework.util.MultiValueMap;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.springframework.test.util.AssertionErrors.fail;
+
+/**
+ * Factory for assertions on the form data parameters.
+ *
+ * <p>An instance of this class is typically accessed via {@link MockRestRequestMatchers#formData()}
+ *
+ * @author Valentin Spac
+ * @since 5.2
+ */
+public class FormDataRequestMatchers {
+
+
+	public RequestMatcher value(String parameter, String... expectedValues) {
+		List<Matcher<? super String>> matcherList = Arrays.stream(expectedValues)
+				.map(Matchers::equalTo)
+				.collect(Collectors.toList());
+
+		return this.value(parameter, matcherList);
+	}
+
+	@SafeVarargs
+	@SuppressWarnings("varargs")
+	public final RequestMatcher value(String parameter, Matcher<? super String>... matchers) {
+		return this.value(parameter, Arrays.stream(matchers).collect(Collectors.toList()));
+	}
+
+	public RequestMatcher value(String parameter, Matcher<Iterable<? extends String>> matchers) {
+		return new AbstractFormDataRequestMatcher() {
+			@Override
+			protected void matchInternal(MultiValueMap<String, String> requestParams) {
+				assertValueCount(parameter, requestParams, 1);
+
+				List<String> values = requestParams.get(parameter);
+				assertThat("Request parameter [" + parameter + "]", values, matchers);
+			}
+		};
```

Comment:
I think these might look better with a lambda + a static method to convert the body to params.